src/window.rs

```diff
@@ -148,8 +148,8 @@ impl WindowBuilder {
                 // resizing the window to the dimensions of the monitor when fullscreen
                 LogicalSize::from_physical(monitor.get_dimensions(), 1.0)
             } else {
-                // default dimensions
-                (1024, 768).into()
+                // use requested or fallback to default dimensions
+                self.window.dimensions.unwrap_or((1024, 768).into())
```

Comment:
I don't understand, this code is already in a `self.window.dimensions.unwrap_or_else(|| { ... })`, how can this change make any difference?

Reply:
Ooops! I didn't notice that this is wrapped in a `unwrap_or_else`. Even though this still fixes my issue. I will try to figure out what is happening here.